### DISCUSSION TITLE
Exporting an RT Plan DICOM from matRad

### DISCUSSION THREAD

**Discussion by oapaza12** [e0404/matRad#767] [Q&A]
Hello everyone,

I have a question regarding matRad. Is it possible to export the plan generated in matRad as a DICOM file, specifically in the RT Plan format?

Thank you for your assistance!

**Comment by wahln** [comment]
Hi, 
it is principally possible on the current [dev](https://github.com/e0404/matRad/tree/dev) branch, at least for photons. Particles are not yet supported, but if it is required (by you?), we may look at it (or help you in implementing it).

The dicom export is implemented in [matRad_DicomExporter](https://github.com/e0404/matRad/tree/dev/matRad/dicom/%40matRad_DicomExporter).

> [!WARNING]  
> We are a little bit hesitant with allowing RTPlan exports (a little bit anxious of unconscious misuse of generated RTPlans in clinical workflows). Thus on the branch the DICOM export of RTPlans is protected by two barriers:  
> 1. Needs to be enabled in the dicom exporter object, e.g., `dcmExporter.enableRtPlanExport = true;`
> 2. An error throwing safeguard needs to be commented out in the corresponding function of the exporter to make sure users know what they are doing!

In the first example, the last lines perform an exemplary dicom export:
https://github.com/e0404/matRad/blob/b25659d7f92cad6fd8b40c63a9db1d1fda956d15/examples/matRad_example1_phantom.m#L146-L154

dicomDir can be every directory, in above example it uses the primary user folder (new concept) stored in the matRad_cfg object (which thus needs to be instantiated), but you can also use any absolute path to your liking.

**Reply by qianyujia** [reply]
Hello, we would be very interested in having this functionality implemented in proton. Could you please let us know how to add the proton support to the DICOMRTplan export? Thank you.

**Reply by wahln** [reply]
we are not explicitly working on this right now.
Still, adding proton support is not per se difficult, but just tedious effort.

In these lines, the DICOM exporter loops over the beams, and then for photons stores the shape information.
https://github.com/e0404/matRad/blob/162bde819f86d7628b208f3ffe673cc13f90152e/matRad/dicom/%40matRad_DicomExporter/matRad_exportDicomRTPlan.m#L155-L333

One could add a switch for protons there and export an RTPlan for protons from the `stf` (contains the geometry) and the weights stored in `resultGUI`.

What needs to be stored could be "reverse engineered" from the DICOM import here:
https://github.com/e0404/matRad/blob/162bde819f86d7628b208f3ffe673cc13f90152e/matRad/dicom/%40matRad_DicomImporter/matRad_importDicomSteeringParticles.m

if you want to look into this, we could certainly help with the implementation

**Reply by qianyujia** [reply]
Thank you for the guidance and we will try. Would it be possible for you to help the implementation for proton? We would greatly appreciate it.

**Comment by oapaza12** [comment]
Hello Niklas Wahl, thank you for your response. Fortunately, I'm working specifically with photons using the developer mode function on the TrueBeam Linac. I'll try to resolve the issue.

**Reply by edwardwang1** [reply]
Hi @oapaza12,

I am also trying to use matRad to export an RTPlan for a Varian TrueBeam. Are you using Eclipse? I think it could be very helpful for both of us to compare notes.

**Reply by oapaza12** [reply]
Hello EdwardWang1,
Absolutely, we can share our notes. Yes, I’m using the Eclipse TPS and executing XML files in developer mode on the TrueBeam linac.

**Reply by oapaza12** [reply]
This is my personal email: oscar.12apaza@gmail.com. Please send me a message so we can discuss our projects in more detail.

**Reply by wahln** [reply]
It would also be really nice to create a bigger discussion topic here on GitHub on this. It seems like that recalculation together with creation of own datasets is less trivial with other machines and algorithms than with the one we used for validation, and there is probably big potential for improvement. I will create something.

**Reply by wahln** [reply]
Opened this thread for shared discussion on this topic:
#772
